text/0000-delete-load.md

```diff
@@ -0,0 +1,48 @@
+# Replace frozen trajectory from file
+
+## Summary
+[summary]: #summary
+
+We propose to add `PoseGraphInterface::DeleteTrajectory`, `PoseGraphInterface::GetTrajectoryStates`, and `MapBuilderInterface::LoadStateFromFile` because these functions are missing to replace a frozen trajectory ("map") at run-time.
+
+## Motivation
+[motivation]: #motivation
+
+Currently, it is not possible to replace a finished (and frozen) trajectory at run-time.
+In fact, the trimmers are the only components that even delete parts of the pose graph.
+When we add these two functions to the public API, the use case of swapping a frozen map at run-time is possible.
+A caller can load a new frozen map, wait for global SLAM to process it (and re-localize in the new map), and delete the old frozen map.
```

Comment:
How do we deal with the possibly different origins of the frozen map coordinate frame?

Reply:
Even when the coordinate frame is correct, there is absolutely no guarantee that re-localization will happen.
Is it a very frequent use case that the replacement frozen map has a different coordinate frame?
In my opinion, this mostly adds complexity, and it is already the caller's responsibility that the replacement trajectory is so similar that localization will be fine, so it'd keep it simple for now and not support different origins.

Reply:
This also makes more sense to me if we think e.g. of the use case of streaming local map updates to a server instance and regularly updating the localization maps on the clients by a new pbstream file. In this case we'd aim to have the old and the new frozen map aligned to each other.

However, there _might_ be situations where it might be problematic, that's why I asked.